aws/table_aws_wellarchitected_lens_review_improvement.go

```diff
@@ -0,0 +1,219 @@
+			},
+			{
+				Name:        "question_id",
+				Description: "The ID of the question.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ImprovementSummary.QuestionId"),
+			},
+			{
+				Name:        "question_title",
+				Description: "The title of the question.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ImprovementSummary.QuestionTitle"),
+			},
+			{
+				Name:        "risk",
+				Description: "The risk for a given workload, lens review, pillar, or question.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ImprovementSummary.Risk"),
+			},
+			{
+				Name:        "improvement_plans",
+				Description: "The improvement plan details.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ImprovementSummary.ImprovementPlans"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ImprovementSummary.QuestionTitle"),
+			},
+		}),
+	}
+}
+
+type ReviewImprovementInfo struct {
+	LensAlias       *string
+	LensArn         *string
+	MilestoneNumber int32
+	WorkloadId      *string
+	types.ImprovementSummary
+}
+
+//// LIST FUNCTION
+
+func listWellArchitectedLensReviewImprovements(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	workload := h.Item.(types.WorkloadSummary)
+
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_lens_review_improvement.listWellArchitectedLensReviewImprovements", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	if d.EqualsQualString("workload_id") != "" {
+		if d.EqualsQualString("workload_id") != *workload.WorkloadId {
+			return nil, nil
+		}
+	}
+
+	for _, lensAlias := range workload.Lenses {
+		// Check for reduce the numbers of API call if lens_alias or lens_arn is provided in query parameter
+		if d.EqualsQualString("lens_alias") != "" {
+			if d.EqualsQualString("lens_alias") != lensAlias {
+				continue
+			}
+		}
+
+		// Limiting the results
+		maxLimit := int32(100)
+		if d.QueryContext.Limit != nil {
+			limit := int32(*d.QueryContext.Limit)
+			if limit < maxLimit {
+				maxLimit = limit
+			}
+		}
+
+		input := &wellarchitected.ListLensReviewImprovementsInput{
+			WorkloadId: workload.WorkloadId,
+			LensAlias:  aws.String(lensAlias),
+			MaxResults: maxLimit,
+		}
+
+		if d.EqualsQuals["milestone_number"] != nil {
+			input.MilestoneNumber = int32(d.EqualsQuals["milestone_number"].GetInt64Value())
+			plugin.Logger(ctx).Debug("aws_wellarchitected_lens_review_improvement.listWellArchitectedLensReviewImprovements", "milestone_number", input.MilestoneNumber)
```

Comment:
Do we get a validation error if the `milestone_number` is more than 100?

https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_ListLensReviewImprovements.html

If not we should return an error saying that the value should be less than 100?

Reply:
Added the condition